pilot/pkg/networking/core/cluster_builder.go

```diff
@@ -353,17 +352,20 @@ func (cb *ClusterBuilder) buildCluster(name string, discoveryType cluster.Cluste
 		// If stat name is configured, build the alternate stats name.
 		if len(cb.req.Push.Mesh.OutboundClusterStatName) != 0 {
 			statPrefix := telemetry.BuildStatPrefix(cb.req.Push.Mesh.OutboundClusterStatName, string(service.Hostname), subset, port, 0, &service.Attributes)
-			if util.IsIstioVersionGE123(cb.proxyVersion) && statPrefix[len(statPrefix)-1:] != constants.StatPrefixDelimiter {
-				statPrefix += constants.StatPrefixDelimiter
-			}
-
-			ec.cluster.AltStatName = statPrefix
+			ec.cluster.AltStatName = util.DelimitedStatsPrefix(statPrefix, cb.proxyVersion)
 		}
 	}
 
 	return ec
 }
 
+func (cb *ClusterBuilder) defaultAltStatName(clusterName string) string {
```

Comment:
I think you caught a bug in my use of the feature flag; if the flag is false, we should never append the delimeter because that flag controls the bootstrap template. So I think all uses of `delimeterAltStatName` should probably be changed to use `DelimitedStatsPrefix`

Reply:
Oh. Yeah. I realized that too and fixed some of the stuff by moving it to use DelimitedStatsPrefix. I thought I moved these also. Let me change. Thank you for catching this

Reply:
Done